.github/workflows/main.yml

```diff
@@ -21,6 +21,24 @@ jobs:
       - name: Check mix format
         run: mix format --check-formatted
 
+  warnings:
+    name: Compile warnings check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Install OTP and Elixir
+        uses: actions/setup-elixir@v1
+        with:
+          otp-version: 23.0.4
+          elixir-version: 1.10.4
+
+      - name: Install dependencies
+        run: mix deps.get
+
+      - name: Check compilation warnings
+        run: mix compile --warnings-as-errors
```

Comment:
Can you add this to the `mix test` step instead?

Reply:
Sure. Added that as a compile step before running tests.